torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,327 @@
+import functools
+from abc import ABC, abstractmethod
+from glob import glob
+from pathlib import Path
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+from PIL import Image
+
+from .utils import _read_pfm, verify_str_arg
+from .vision import VisionDataset
+
+__all__ = ()
+
+_read_pfm_file = functools.partial(_read_pfm, slice_channels=1)
+
+
+class StereoMatchingDataset(ABC, VisionDataset):
+    """Base interface for Stereo matching datasets"""
+
+    _has_built_in_disparity_mask = False
+
+    def __init__(self, root: str, transforms: Optional[Callable] = None):
+        """
+        Args:
+            root(str): Root directory of the dataset.
+            transforms(callable, optional): A function/transform that takes in Tuples of
+                (images, disparities, valid_masks) and returns a transformed version of each of them.
+                images is a Tuple of (``PIL.Image``, ``PIL.Image``)
+                disparities is a Tuple of (``np.ndarray``, ``np.ndarray``) with shape (1, H, W)
+                valid_masks is a Tuple of (``np.ndarray``, ``np.ndarray``) with shape (H, W)
+                In some cases, when a dataset does not provide disparities, the ``disparities`` and
+                ``valid_masks`` can be Tuples containing None values.
+                For training splits generally the datasets provide a minimal guarantee of
+                images: (``PIL.Image``, ``PIL.Image``)
+                disparities: (``np.ndarray``, ``None``) with shape (1, H, W)
+                Optionally, based on the dataset, it can return a ``mask`` as well:
+                valid_masks: (``np.ndarray | None``, ``None``) with shape (H, W)
+                For some test splits, the datasets provides outputs that look like:
+                imgaes: (``PIL.Image``, ``PIL.Image``)
+                disparities: (``None``, ``None``)
+                Optionally, based on the dataset, it can return a ``mask`` as well:
+                valid_masks: (``None``, ``None``)
+        """
+        super().__init__(root=root)
+        self.transforms = transforms
+
+        self._images = []  # type: ignore
+        self._disparities = []  # type: ignore
```

Comment:
Do we really need to add `type: ignore` here? I'd suggest to type the rest, or not type anything at all (which is by far my prefered way)

Reply:
I've left the `#type: ingore` because mypy starts throwing errors about empty lists since it cannot infer their dtype.

`torchvision/datasets/_stereo_matching.py:48: error: Need type annotation for "_images" (hint: "_images: List[<type>] = ...")  [var-annotated]
            self._images = []
`

`torchvision/datasets/_stereo_matching.py:49: error: Need type annotation for "_disparities" (hint: "_disparities: List[<type>] = ...")  [var-annotated]
            self._disparities = []
`